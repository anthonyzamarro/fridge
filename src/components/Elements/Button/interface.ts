export interface ButtonProps {
  text: string;
  handler: () => void;
}
