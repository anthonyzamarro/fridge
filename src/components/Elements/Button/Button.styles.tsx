import styled from 'styled-components';

export const StyledButton = styled.input`
  background-color: #d2d3d2;
`;
