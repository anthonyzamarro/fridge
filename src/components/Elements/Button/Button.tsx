// interface
import { ButtonProps } from './interface';
// styles
import { StyledButton } from './Button.styles';

const Button = ({ text, handler }: ButtonProps) => (
  <StyledButton onClick={handler}>{text}</StyledButton>
);

export default Button;
