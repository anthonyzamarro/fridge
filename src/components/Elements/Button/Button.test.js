import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

test('button contains text', () => {
  const component = renderer.create(<Button text="click me test" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
