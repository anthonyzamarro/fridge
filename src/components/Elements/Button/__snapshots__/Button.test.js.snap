// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button contains text 1`] = `
<input
  className="sc-bdvvtL bdfzay"
>
  click me test
</input>
`;
